hawc/apps/lit/views.py

```diff
@@ -484,20 +484,37 @@ def get_success_url(self):
         return reverse_lazy("lit:overview", args=[self.assessment.pk])
 
 
-class RefListExtract(BaseList):
-    parent_model = Assessment
-    model = models.Reference
-    crud = "Update"  # update-level permission required despite list-view
+class RefListExtract(TeamMemberOrHigherMixin, MessageMixin, FormView):
     template_name = "lit/reference_extract_list.html"
     breadcrumb_active_name = "Prepare for extraction"
+    model = Assessment
+    form_class = forms.BulkReferenceStudyExtractForm
+    success_message = "Items created!"
 
-    def get_queryset(self):
-        return self.model.objects.get_references_ready_for_import(self.assessment)
+    def get_assessment(self, request, *args, **kwargs):
+        return get_object_or_404(self.model, pk=kwargs["pk"])
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update(
+            assessment=self.assessment,
+            reference_qs=models.Reference.objects.get_references_ready_for_import(self.assessment),
+        )
+        return kwargs
 
     def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context["breadcrumbs"].insert(2, lit_overview_breadcrumb(self.assessment))
-        return context
+        kwargs.update(
+            breadcrumbs=lit_overview_crumbs(self.request.user, self.assessment, "Reference upload"),
+            object_list=models.Reference.objects.get_references_ready_for_import(self.assessment),
```

Comment:
we get this queryset twice, once in the form_kwargs, and once in the context_data. Is there a way that we can get it once and use it in both places?  You might be able to change this line to 

```python
kwargs.update(
    breadcrumbs=lit_overview_crumbs(self.request.user, self.assessment, "Reference upload"),
    object_list=context['form'].fields['reference_ids'].queryset,
)
```

Not sure if form is in the context yet; it might be a `self.form` or in some other location?  If it's not, then we should add it somehow to this class, so we're not doing a db double-hit.

Reply:
I got it! I had to move object_list after calling the super get_context_data